src/runtime/hexagon_remote/Makefile

```diff
@@ -24,13 +24,17 @@
 #
 
 SDK_ROOT ?= ${HOME}/Qualcomm
-HEXAGON_SDK_ROOT ?= ${SDK_ROOT}/Hexagon_SDK/3.3.3
-HEXAGON_TOOLS_ROOT ?= ${HEXAGON_SDK_ROOT}/tools/HEXAGON_Tools/8.1.05
+HEXAGON_SDK_ROOT ?= ${SDK_ROOT}/Hexagon_SDK/4.3.0.0
+# Need SDK >= 4.2.0.2 for qaic "halide_hexagon_remote_skel_invoke_qaic_version"
```

Comment:
What is this referring to? I only see one instance of this string in this PR (here).

Reply:
That is a symbol that is automatically generated by the newer qaic compiler.  It can be seen in the generated skel source:

[vm-dpalermo-aus Halide/src/runtime/hexagon_remote/bin/src]$ grep ver *skel.c
__attribute__ ((visibility("default"))) uint32_t halide_hexagon_remote_skel_invoke_qaic_version = 10041;

The purpose of it is to allow the DSP dispatch to be able to verify the skel shared object was built with a "new enough" version of qaic.  This was done so it is possible to determine at runtime if a given skel file was generated with a qaic that contained fixes for all critical security fixes (such as the qaic in SDK 4.2.0.2 contains).

Also, the dispatch logic for all devices >sm8350 will only run skel shared objects that have this symbol present and has a value >=10041 (but this won't affect existing Pixel devices).  Older devices (Pixel 3*,4*,5) with the current dispatch obviously won't even be looking for this symbol....but by using the latest qaic, will still have the security fixes...even though they won't be checking for them.

Reply:
I think this comment might be a bit out of place. I think it should either be removed or elaborated on. I assume this is documented in the Hexagon SDK, perhaps we should just refer to that documentation here?

Reply:
I don't see anything in the SDK docs that goes into the above detail.
 
How about I change the comment to:
`# Need SDK >= 4.2.0.2 for qaic version 01.00.41 for critical security fixes`

Or I can just remove the comment.

Let me know what you think.

Reply:
Adding a comment that `halide_hexagon_remote_skel_invoke_qaic_version` is autogenerated by qaic is essential, I think

Reply:
Why? How does it help the user? 

From dpalermo's comment above, it seems like the thing that really should be commented here is this:

# Devices >sm8350 will only run skel shared objects that have been built with SDK version 4.2.02 or newer.

But really, this should be documented in more detail in the Hexagon SDK, and we should be referring to that documentation instead.

Reply:
I think the reason I'm not finding anything specific in the docs about this is that the latest device described in Hexagon_SDK/4.3.0.0/docs is the SM8350.  More likely will be documented in the next SDK release.